src/firewalld.py

```
@@ -218,94 +220,104 @@ def _check_firewalld_version(self) -> bool:
-                    ])
+                    success, _, stderr = self._run_firewall_cmd(
+                        [
+                            "--permanent",
+                            f"--zone={self.zone_name}",
+                            f"--add-rich-rule={default_rule}",
+                        ]
+                    )
                     if not success:
-                        self.logger.warning(f"Failed to add {self.default_action.upper()} rule for {port}/{protocol} ({family}): {stderr}")
+                        self.logger.warning(
+                            f"Failed to add {self.default_action.upper()} rule for {port}/{protocol} ({family}): {stderr}"
```

Ruff (0.15.13)

[warning] Logging statement uses f-string

(G004)